lib/index.js

```diff
@@ -90,17 +94,60 @@ MaplibreGeocoder.prototype = {
     clearAndBlurOnEsc: false,
     clearOnBlur: false,
     getItemValue: function (item) {
-      return item.place_name;
+      return typeof item === "string" ? item : item.place_name;
     },
     render: function (item) {
-      var placeName = item.place_name.split(",");
-      return (
-        '<div class="mapboxgl-ctrl-geocoder--suggestion maplibregl-ctrl-geocoder--suggestion"><div class="mapboxgl-ctrl-geocoder--suggestion-title maplibregl-ctrl-geocoder--suggestion-title">' +
-        placeName[0] +
-        '</div><div class="mapboxgl-ctrl-geocoder--suggestion-address maplibregl-ctrl-geocoder--suggestion-address">' +
-        placeName.splice(1, placeName.length).join(",") +
-        "</div></div>"
-      );
+      // Render as a suggestion
+      if (typeof item === "string") {
+        var suggestionName = item.split(",");
+        var indexOfMatch = suggestionName[0]
+          .toLowerCase()
+          .indexOf(this.query.toLowerCase());
+        var lengthOfMatch = this.query.length;
+        var beforeMatch = suggestionName[0].substring(0, indexOfMatch);
+        var match = suggestionName[0].substring(
+          indexOfMatch,
+          indexOfMatch + lengthOfMatch
+        );
```

Comment:
Can we get this logic reviewed with the service team.

Reply:
What part of this logic are you worried about with the service team? Most of this logic is just to bold the substring of the result that matches the suggestion search text.

The only part here what comes into play would be the separation at the first comma to split the lines in the UI. We've been doing this already in the `searchResults` the entire time (line 135 here, but on `main` it's line [96](https://github.com/maplibre/maplibre-gl-geocoder/blob/af838e5e20c3ba18108c16ed886e4d294037b597/lib/index.js#L96)).

Reply:
Yes this one
>the separation at the first comma to split the lines in the UI

I just want to make sure that this assumption is accurate.

Reply:
`,` is the decoupling item from our service team, however, there are some cases where this would cause odd behavior in non-us country's addresses, such as in these screenshots:
![image](https://user-images.githubusercontent.com/16496746/153490491-71a8910d-efd9-4ac7-be54-79b7da0fc584.png)
![image](https://user-images.githubusercontent.com/16496746/153490494-4a6775bc-1b69-4994-a297-74f9e30ef466.png)

The only other option here is to not separate and use two lines at all, however, we'd be able to show less information about the location. If developers want to overwrite this behavior, they can do so by passing a custom `render` function to the geocoder constructor options object when called.